service/server/server.go

```diff
@@ -42,30 +57,60 @@ func NewOracleServer(o servicetypes.Oracle, logger *zap.Logger) *OracleServer {
 	}
 	os.Closer = sync.NewCloser().WithCallback(func() {
 		// if the server has been started, close it
-		if os.srv != nil {
-			os.srv.GracefulStop()
+		if os.httpSrv != nil {
+			ctx, cf := context.WithTimeout(context.Background(), DefaultServerShutdownTimeout)
+			_ = os.httpSrv.Shutdown(ctx)
+			cf()
 		}
 	})
 
 	return os
 }
 
+// routeRequest determines if the incoming http request is a grpc or http request and routes to the proper handler
+func (os *OracleServer) routeRequest(w http.ResponseWriter, r *http.Request) {
+	if r.ProtoMajor == 2 && strings.HasPrefix(
+		r.Header.Get("Content-Type"), "application/grpc") {
```

Comment:
I took this from the docs https://pkg.go.dev/google.golang.org/grpc#Server.ServeHTTP and it works.